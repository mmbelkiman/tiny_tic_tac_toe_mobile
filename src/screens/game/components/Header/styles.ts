import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'column',
  },
  backText: {
    paddingLeft: 20,
    pointerEvents: 'none',
    fontSize: 50,
    color: '#999',
  },
});
