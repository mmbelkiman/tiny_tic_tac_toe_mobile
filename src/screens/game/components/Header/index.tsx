import {Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import styles from './styles';

interface HeaderProps {
  onPressBack: () => void;
  testID?: string;
}

const Header: React.FC<HeaderProps> = ({onPressBack, testID = ''}) => {
  return (
    <View style={styles.container}>
      <View>
        <TouchableOpacity
          testID={`Header${testID}.TouchableOpacity.back`}
          onPress={onPressBack}>
          <Text style={styles.backText}>{'←'}</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Header;
