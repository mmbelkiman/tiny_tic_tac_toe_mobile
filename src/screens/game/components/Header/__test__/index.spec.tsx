import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Header from '../index';

describe('NumberInput component', () => {
  it('button works', () => {
    const onPressBackMock = jest.fn();
    const {getByTestId} = render(<Header onPressBack={onPressBackMock} />);

    const backButton = getByTestId('Header.TouchableOpacity.back');
    fireEvent.press(backButton);

    expect(onPressBackMock).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const onPressBackMock = jest.fn();
    const {toJSON} = render(<Header onPressBack={onPressBackMock} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
