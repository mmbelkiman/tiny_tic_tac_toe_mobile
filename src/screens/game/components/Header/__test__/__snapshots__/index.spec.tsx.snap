// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput component renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Header.TouchableOpacity.back"
    >
      <Text
        style={
          {
            "color": "#999",
            "fontSize": 50,
            "paddingLeft": 20,
            "pointerEvents": "none",
          }
        }
      >
        ‚Üê
      </Text>
    </View>
  </View>
</View>
`;
